### PR TITLE
Add notification webhooks

### PR DESCRIPTION
**Pull request by ghost**
This adds support for notifications from openbadger, allowing emails to be sent out when badges are issued directly from openbadger.

### CODE REVIEW

**Comment by stenington** [comment]
@christensenep You mentioned this is live but unmerged... I'm happy to try to review but I don't know what the webhooks are for. Is there somewhere I can read up on that?

**Comment by ghost** [comment]
There isn't any documentation right now other than openbadger's documentation simply mentioning their existence.  I'll modify that README to explain what behavior openbadger is expecting from the endpoint you provide.

Briefly, though, the idea is that when you issue a badge to someone in openbadger, it notifies the client site (CSOL-site, or openbadgges-badges, or whatever) that there is a new claim code waiting for someone, and the client site is responsible for handling that situation appropriately.  Typically this involves emailing the person and/or immediately claiming the badge via the openbadger API.

It's a little weird that it's up to the client site to claim the badge when it was actually issued from directly within openbadger, but this was done for CSOL in order to avoid problems with underage learners.

**Comment by stenington** [comment]
I guess given that this is out there and functioning, it must be doing the right thing. 

It's tempting to suggest refactoring out the commonality between the `/notify/claim` POST handler and the existing claim stuff, but it seems like it has just enough slight differences that that might not be worth it, which seems to happen a lot with route handling.

**Comment by ghost** [comment]
That's a really good point, though.  I'd prefer not to mess with it today to that extent if that's ok, but I think that would definitely be an improvement, and I think the minor differences could probably be handled without too many funky gymnastics.

File: app/lib/openbadger-hooks.js [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+const JWT_SECRET = process.env['OPENBADGER_SECRET'];
+
+module.exports = function makeOpenbadgerHooks(openbadger) {
+  function respondWithForbidden(res, reason) {
+    return res.send(403, { status: 'forbidden', reason: reason });
+  }
+
+  function respondWithError(res, err) {
+    return res.send(500, { status: 'error', error: 'An error occurred while attempting to handle this claim notification.' });
+  }
```
Comment (andrewhayward) [inline app/lib/openbadger-hooks.js:15]: Is `err` supposed to be used at all here? Seems somehow redundant.

File: app/lib/openbadger-hooks.js [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    define: function defineRoutes(app) {
+      app.post('/notify/claim', auth, function(req, res, next) {
+        var claimCode = req.body.claimCode;
+        var email = req.body.email;
+
+        if (req.body.isTesting)
+          return res.send(200, { status: 'ok' });
+
+        if (!claimCode)
+          return res.send(500, { status: 'error', error: 'No claimCode provided' });
```
Comment (andrewhayward) [inline app/lib/openbadger-hooks.js]: Seems like you should be making use of your `respondWithError` here, but passing the supplied error along.

**Comment by andrewhayward** [comment]
Given that this is already working live, I'm not too fussed about the minor error-related stuff. If you wanted to fix it, by all means do so, but I don't think it's a big enough issue to really hold anything up. There also doesn't seem to be a lot of point in defining those `respondWith...` and `auth` method inside the `exports` function, but again, it's really not that big a deal.

So, I guess what I'm getting at is that I'm happy for this to be merged, with the caveat that an issue be opened to review any refactoring that might need to take place.

**Comment by ghost** [comment]
I'm actually going to fix that error stuff right now, that's a good catch.  Just testing it now, will push shortly.

**Pull request merged**